code/datums/components/chasm.dm

```diff
@@ -35,13 +35,17 @@ GLOBAL_LIST_INIT(chasm_storage, list())
 	))
 
 /datum/component/chasm/Initialize(turf/target)
-	RegisterSignal(parent, COMSIG_ATOM_ENTERED, PROC_REF(Entered))
+	RegisterSignal(parent, SIGNAL_ADDTRAIT(TRAIT_CHASM_STOPPED), PROC_REF(on_chasm_stopped))
+	RegisterSignal(parent, SIGNAL_REMOVETRAIT(TRAIT_CHASM_STOPPED), PROC_REF(on_chasm_no_longer_stopped))
 	target_turf = target
-	START_PROCESSING(SSobj, src) // process on create, in case stuff is still there
+	RegisterSignal(parent, COMSIG_ATOM_ENTERED, PROC_REF(entered))
+	RegisterSignal(parent, COMSIG_ATOM_EXITED, PROC_REF(exited))
+	RegisterSignal(parent, COMSIG_ATOM_INITIALIZED_ON, PROC_REF(initialized_on))
+	//allow catwalks to give the turf the CHASM_STOPPED trait before dropping stuff
+	addtimer(CALLBACK(src, PROC_REF(drop_stuff)), 0)
```

Comment:
I really hate 0 time timers but I guess I can't think of a good alternative here if what you're waiting for is "a different object to init and give our turf a trait"
No chance this would work on `LateInitialize`? It presumably shouldn't be a problem after mapload?

Reply:
It won't do alas. The turf traits would have to be (re)added before the new component has finished initializing, and keep in mind signals won't work until the turf has finished initializing in the case of ChangeTurf.